### PR TITLE
[RL] support weight reload for low-bit rollout

### PR DESCRIPTION
**Pull request by AniZpZ**
## Motivation

add support for weight reload and quantization on the fly to support low-bit rollout 
easy access to [Flash-RL](https://github.com/yaof20/Flash-RL) with sglang as rollout engine

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Comment by Wilboludriver** [comment]
W B Chart 2025_12_5 15_23_19
W B Chart 2025_12_5 15_23_55
W B Chart 2025_12_5 15_24_28
W B Chart 2025_12_5 15_25_05
W B Chart 2025_12_5 15_31_33



**Experimental Details**
Model: [Qwen2.5-14B-Instruct](https://huggingface.co/Qwen/Qwen2.5-14B-Instruct)
Training Recipe: DAPO 
Configuration: 
  - Training Dataset: DAPO-Math-17k
  - Quantization Scheme: Per-channel fp8
  - Validation: GSM8K
  - Prompt batch size 32, n=16.
  - Rollout batch size: 32*3*16
  - Train_batch_size & ppo_mini_batch_size 32
  - Token-level TIS, C=2
  - 8*H20. veRL
  
 Observations:
 With TIS the mismatch between per-channel quantized FP8 rollout and native BF16 rollout gets alleviated, which can be also inferred from the rewards curves.
 
 Outlook:
 Current implementaion contains sub-optimized data movement during the weight reloading phase, which is now under fix for higher troughput. For the quantization sheme, per-channel fp8 dynamic quantization significantly destroys the process of thinking of Qwen3-8B during training due to our experiments. We are now implementing dynamic blockwise fp8 rollout, which showcases higher robustness acc. to other reports. @AniZpZ

**Review by zhaochenyang20 — APPROVED** [review]
Great job

**Review by ispobock — APPROVED** [review]

**Comment by CSWYF3634076** [comment]
Can this feature be run on MoE model？
